### PR TITLE
fix pcre2 version

### PR DESCRIPTION
**Pull request by ihji**
the latest version is 7.5.2 (https://github.com/camlp5/pcre2-ocaml)

### CODE REVIEW

**Comment by aryx** [comment]
I thought we were using the pcre2 included under libs/pcre2?
Did you run `git submodule update --init` before running make setup?

File: dune-project [hunk 1]
```diff
@@ -104,7 +104,7 @@ a few other projects developed at r2c.
     (yojson (>= "1.7.0"))
     (re (>= "1.10.4"))
     (pcre (>= "7.5.0" ))
-    (pcre2 (>= "7.5.3" ))
```
Comment (aryx) [inline dune-project:107]: @kopecs we should probably remove this no? pcre2 is "vendored" under libs/, so no
need to install it via opam no?

**Review by aryx — CHANGES_REQUESTED** [review]

**Comment by kopecs** [comment]
The version is from https://github.com/semgrep/pcre2-ocaml which includes bugfixes not in upstream.

File: dune-project (hunk 1 above)
Comment (kopecs) [reply dune-project:107]: Hmm, we still depend on it, so I would assume we want it present here, no?

**Comment by ihji** [comment]
> The version is from https://github.com/semgrep/pcre2-ocaml which includes bugfixes not in upstream.

Can 7.5.3 be installed via the opam package repository? I've seen 
```
[ERROR] Package conflict!
  * Missing dependency:
    - pcre2 >= 7.5.3
    no matching version
```
when I `make setup` in one of my local copies of the pro repo, so I assumed that 7.5.3 doesn't exist. However, I'm fine with leaving this as is if the version annotation is not incorrect.

File: dune-project (hunk 1 above)
Comment (aryx) [reply dune-project:107]: no, this is used to install external packages.
It's an opam thing (it's in a dune file, but dune just use it to autogenerate the semgrep.opam for OPAM).

**Comment by aryx** [comment]
probably make setup fail because you didn't have the submodule populated so opam could not
find the pcre2.opam locally so it went to fetch the one from OPAM repo and then got the conflict.

**Comment by aryx** [comment]
probably simpler to remove prcre2 from dune-project (and semgrep.opam) so it does not even try.

**Comment by aryx** [comment]
superseded by https://github.com/semgrep/semgrep/pull/10207

**Pull request closed without merging**